src/controllers/loginController.js

```diff
@@ -0,0 +1,72 @@
+let passport = require('passport')
+
+let db = require('../db')
+let ApiError = require('../lib/apiErrors')
+let $ = require('../lib/controllerHelpers')
+
+exports.postLogin = (req, res, next) => {
+  // req.assert('email', 'Email is not valid').isEmail()
+  // req.assert('password', 'Password cannot be blank').notEmpty()
+
+  let errors = req.validationErrors()
+  if (errors) {
+    return next(new ApiError.FailedToLogin(errors))
+  }
+
+  passport.authenticate('local', (err, user, info) => {
+    if (!user || err) {
+      return next(new ApiError.FailedToLogin(err))
+    }
+
+    req.logIn(user, (loginErr) => {
+      if (loginErr) {
+        return next(new ApiError.FailedToLogin(loginErr))
+      }
+      res.json({
+        state: req.isAuthenticated(),
+        user: user
+      })
+    })
+  })(req, res, next)
+}
+
+exports.logout = (req, res) => {
+  req.logout()
+  res.locals.user = null
+}
+
+exports.postSignup = (req, res, next) => {
+  let params = $.requireParams([
+    'email',
+    'password',
+    'universityId'
+  ], req.body)
+
+  let errors = req.validationErrors()
+  if (errors) {
```

Comment:
The `#validationErrors()` function is a synchronous function. There is an asynchronous version that may be worth updating to in the future, but for the sake of simplicity, I chose this one for now.